beamform/craftcor_tab.py

```diff
@@ -131,25 +163,112 @@ def do_f_tab(self, corr, iant):
         total_delay_samp = framediff_samp
         whole_delay = int(np.round(total_delay_samp))
 
-        data_out = np.zeros(
-            (corr.nint, corr.nfine_chan, corr.npol_in), dtype=np.complex64
-        )
+
+        #############################################################
+        # Where the loading of the data happens
+        #
+        #############################################################
+
+
         nfine = corr.nfft - 2 * corr.nguard_chan
 
         nsamp = corr.nint * corr.nfft
+        nchan_coarse = len(corr.freqs)
 
         # time-dependent geometric delays
         # np.linspace(0, 1, nsamp) == time in units of integrations
         geom_delays_us = (
             geom_delay_us
-            + geom_delay_rate_us * np.linspace(0, 1, nsamp)
+            + geom_delay_rate_us * np.linspace(0, 1, nsamp, dtype = np.float32)
             - fixed_delay_us
         )
         np.save("TEMP_geom_delays_us.npy", geom_delays_us)
         del geom_delays_us
         geom_delays_us = np.load("TEMP_geom_delays_us.npy", mmap_mode="r")
 
         sampoff = whole_delay + corr.abs_delay
+
+        # update sampoff and nsamp based on mjd and DM for better cropping
+        if (mjd is not None) and (DM is not None):
+            # perform rough pulse cropping
+            print(f"Start MJD of buffer: {self.mjdstart}")
+            print(f"Start MJD of cand pulse: {mjd}")
+            pulse_offset_samp = int((mjd - self.mjdstart)*8.64e10)
+
+            # measure DM sweep
+            kDM = 4149.377593
+            DM_sweep_samp = int(abs(kDM * DM * 1e6 * (1/(min(corr.freqs)**2) - 1/(max(corr.freqs)**2))))
```

Comment:
This should really be called sweep_microsec, rather than sweep_samp, since it isn't samples.